pinot-plugins/pinot-stream-ingestion/pinot-kafka-base/src/main/java/org/apache/pinot/plugin/stream/kafka/KafkaJSONMessageDecoder.java

```diff
@@ -18,62 +18,13 @@
  */
 package org.apache.pinot.plugin.stream.kafka;
 
-import com.fasterxml.jackson.core.type.TypeReference;
-import com.fasterxml.jackson.databind.JsonNode;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import java.util.Arrays;
-import java.util.Map;
-import java.util.Set;
-import org.apache.pinot.spi.data.readers.GenericRow;
-import org.apache.pinot.spi.data.readers.RecordExtractor;
-import org.apache.pinot.spi.plugin.PluginManager;
-import org.apache.pinot.spi.stream.StreamMessageDecoder;
-import org.apache.pinot.spi.utils.JsonUtils;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+
+import org.apache.pinot.plugin.inputformat.json.StreamJSONMessageDecoder;
 
 
 /**
- * An implementation of StreamMessageDecoder to read JSON records from a stream.
+ * This class has been kept for backward compatability. Use @see `org.apache.pinot.plugin.inputformat.json.StreamJSONMessageDecoder` for future use cases.
```

Comment:
Please mark the class with `@Deprecated`

Also, the class is used in configuration, right? So, it is good to mark this PR as needing release notes, so that users can change the configuration to point to the newer class. Do say that we will remove this class eventually.

Reply:
Resolved.